sync:
  interval_ms: 20
  overflow: 0
heartbeat:
  rate: 20
  msg: "77f#05"

defaults:
  eds_pkg: cob_hardware_config
  eds_file: "robots/common/Elmo.dcf"
  dcf_overlay:
    "6083": "1000000"
    "6084": "1000000"
    "60C5": "1000000"
    "60C6": "1000000"
    "6099sub1": "100000"
    "6099sub2": "10000"
    "1016sub1" : "0x7F0064" # heartbeat timeout of 100 ms for master at 127

drive_wheel: &drive_wheel
  dcf_overlay: # "ObjectID" : "ParameterValue" (both as strings)
    "6098": "35" # homing method
  vel_from_device: "p1 != p1 || obj2041 == t1 ? 0 : deg2rad(1000*norm(obj6064-p1,-36000000,36000000)/norm(obj2041-t1,0,2^32)), p1=p0, p0 = obj6064, t1 = t0, t0 = obj2041"

nodes:
  ##Wheel 1
  f_caster_rotation_joint:
    id: 1
    dcf_overlay: # "ObjectID" : "ParameterValue" (both as strings)
      "6098": "19" # homing method
      "607C": "0" # home offset
  f_caster_r_wheel_joint:
    <<: *drive_wheel
    id: 2

  ##Wheel 2
  bl_caster_rotation_joint:
    id: 3
    dcf_overlay: # "ObjectID" : "ParameterValue" (both as strings)
      "6098": "19" # homing method
      "607C": "0" # home offset
  bl_caster_r_wheel_joint:
    <<: *drive_wheel
    id: 4

  ##Wheel 3
  br_caster_rotation_joint:
    id: 5
    dcf_overlay: # "ObjectID" : "ParameterValue" (both as strings)
      "6098": "19" # homing method
      "607C": "0" # home offset
  br_caster_r_wheel_joint:
    <<: *drive_wheel
    id: 6

